root/static/scripts/edit/externalLinks.js

```diff
@@ -508,11 +508,15 @@ export class ExternalLinksEditor
     }
   }
 
+  getOldLink(relationship: StrOrNum | null): ?LinkStateT {
+    const oldLinks = this.getOldLinksHash();
+    return oldLinks.get(String(relationship));
+  }
```

Comment:
I'd prefer not to call `getOldLinksHash` in here if possible, since it has to recompute it each time this is called, which appears to be inside a loop (`links.some`).

Reply:
This is how it was written before. Do you mean we should simply `find` the old link by `relationship`?

Reply:
If we can load oldLinks once, then just use the same oldLinks constant repeatedly, that would seem ideal. I assume oldLinks do not change anyway?

Reply:
It was bad before too, and now it's done in two places. :)

Using `find` could work, or possibly caching `oldLinks` in the constructor.

Reply:
I'll go with caching, it would be useful for implementing highlights in the future.

Reply:
Yeah that should be fine if `this.props.initialLinks` is constant.  Just not if the component renders again with different a different `initialLinks` prop, since the constructor is only run when the component mounts.